platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/MerkleDbDataSource.java

```diff
@@ -1180,41 +1154,44 @@ private void runWithSnapshotExecutor(
         }
     }
 
-    /** Write all internal records hashes to internalHashStore */
-    private void writeInternalRecords(final long firstLeafPath, final Stream<VirtualInternalRecord> internalRecords)
+    /**
+     * Write all hashes to hashStore
+     */
+    private void writeHashes(final long maxValidPath, final Stream<VirtualHashRecord> pathHashRecords)
             throws IOException {
-        if ((internalRecords == null) || (firstLeafPath <= 0)) {
+        if ((pathHashRecords == null) || (maxValidPath <= 0)) {
             // nothing to do
             return;
         }
 
-        if (hasDiskStoreForInternalHashes) {
-            internalHashStoreDisk.startWriting(0, firstLeafPath - 1);
+        if (hasDiskStoreForHashes) {
+            hashStoreDisk.startWriting(0, maxValidPath);
         }
 
         final AtomicLong lastPath = new AtomicLong(INVALID_PATH);
-        internalRecords.forEach(rec -> {
-            assert rec.getPath() > lastPath.getAndSet(rec.getPath()) : "Path should be in ascending order!";
+        pathHashRecords.forEach(rec -> {
+            final long path = rec.path();
```

Comment:
Looks redundant (only used in the assertion below).

Reply:
Ok, I'll inline it.